content_scripts/storageLoader.js

```diff
@@ -219,6 +219,46 @@ function listenForBackgroundMessages() {
     });
 }
 
+/**
+ * General method to scrape DOM inside html
+ * @param url   The url consisting the DOM element
+ * @param cssSelector   CSS selector to find the DOM element
+ * @param all   Get all DOM elements matching the selector or just the first one (default false)
+ * @returns {Promise<NodeListOf<Element>|Element>}  Array of dom elements in case sent with all param "true"
+ *                                                   otherwise just one DOM element
+ */
+async function scrapeWebsiteDOM(url, cssSelector, all = false) {
+    let parser = new DOMParser()
+    let response = await fetch(url)
+    const contentType = response.headers.get('Content-Type')
+    const charsetStartIndex = contentType.lastIndexOf("charset=")
+    let htmlDoc
+    if (charsetStartIndex !== -1) {
+        const charSet = contentType.substring(charsetStartIndex + 8)
```

Comment:
Why 8? maybe add a documentation  :)

Reply:
8 as the string length "charset=",
I'll change it to be more clear and add some comment if it's not clear enough